test/unit/location.d

```diff
@@ -0,0 +1,35 @@
+module location;
+
+import dmd.lexer;
+import dmd.globals : Loc;
+
+unittest
+{
+    string text = "int a b xyz\nfloat\n\nchar a=b + c[11];";
```

Comment:
I expect all tokens to be tested, not just a few. I also expect edge cases to be test, like:

* The first token in the source code
* The last token (i.e. last byte) in the source code
* Test the location of a token after Windows line endings
* Test the location of a token after Mac line endings (if DMD supports those)
* Test the location of a token after Unix line endings
* The end of file token
* A token after the `#line` token
* A token after a comment (all variants of comments)
* Characters and string tokens with multibyte characters

Here's an example of how I've implemented tests for all tokens:

https://github.com/jacob-carlborg/ddc/commit/32d5fe37be211065558f6b44733f059ec9f10c13#diff-ca42ad79bd703d8bea0e34b342881ac9

It's missing some of the above edge cases, but it should provide you with a pretty good starting point.

You can even improve on the above test setup, by instead of storing the tests in an array, store them in an associative array, where the keys are the tokens and the values are the tests. Then you can use loop over the enum members of the `TOK` enum and access the values in the associative array. This will guarantee that if a token is added, it won't get missed in the test because the test will fail.